keras_eval/utils.py

```diff
@@ -113,6 +113,67 @@ def get_concept_items(concepts, key):
     return [concept[key] for concept in concepts]
 
 
+def read_dictionary(dictionary_path):
+    if os.path.exists(dictionary_path):
+        with open(dictionary_path, 'r') as dictionary_file:
+            dict = json.load(dictionary_file)
+    else:
+        raise ValueError('Error: invalid dictionary path' + str(dictionary_path))
+    return dict
+
+
+def create_training_json(train_dir, output_json_file):
+    '''
+            Checks if evaluation concepts are unique
+            Args:
+                train_dir: The location where you have the training directory
+                output_json_file: The output file name and path e.g.: ./dictionary.json
+
+            Returns:
+                True, if there are no repeat concepts, else raises error
+    '''
+    dict = []
+    train_concepts = get_default_concepts(train_dir)
+    for i in range(len(train_concepts)):
+        dict.append({"class_index": i, "class_name": train_concepts[i]["label"], "group": train_concepts[i]["label"]})
+    with open(output_json_file, 'w') as file_obj:
+        json.dump(dict, file_obj)
+
+
+def check_concept_unique(concept_dict):
+    '''
+        Checks if evaluation concepts are unique
+        Args:
+            concept_dict: Dictionary that contains class_id, train_concepts and groups
+        Returns:
+            True, if there are no repeat concepts, else raises error
+    '''
+    concept_class_name_dict = {}
+    for concept_dict_item in concept_dict:
+        if concept_dict_item['class_name'] in concept_class_name_dict:
+            raise ValueError("Concept has been repeated:", concept_dict_item['class_name'])
+        else:
+            concept_class_name_dict[concept_dict_item['class_name']] = 1
```

Comment:
Maybe a best way to avoid storing this `1`?

Reply:
Could not find a better way of doing in O(1).